### PR TITLE
fix: Add type hints for secret manager snippets

### PR DESCRIPTION
**Pull request by glasnt**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: secretmanager/snippets/add_secret_version.py
Comment (glasnt) [inline secretmanager/snippets/add_secret_version.py:45]: mypy saw this as a type mismatch because of the self-assigning type change. Renaming the variable helps with clarity.

File: secretmanager/snippets/quickstart.py
Comment (glasnt) [inline secretmanager/snippets/quickstart.py:68]: I had issues trying to use `str | None` and have valid mypy tests, so I found just providing defaults here helped. I don't _think_ this quickstart is called directly, but we'll see if tests pass 🤞

Comment (engelke) [reply secretmanager/snippets/quickstart.py:68]: I had something similar with defaults. I had to change some defaults from None to "falsy" values of other types, like "", {}, []... This looks really good (and defaulting them to None doesn't seem to have been a good idea anyway).

File: secretmanager/snippets/snippets_test.py
Comment (engelke) [inline secretmanager/snippets/snippets_test.py:69]: Is Generator right here? It looks like it's yielding a str to me. What am I missing?

File: secretmanager/snippets/snippets_test.py
Comment (engelke) [inline secretmanager/snippets/snippets_test.py:87]: Same question about Generator here.

**Review by kweinmeister — APPROVED** [review]

**Review by engelke — APPROVED** [review]
Wow, what a PR! Looks good. I had a couple of questions about the Generator type. If they're right, please add a note helping me understand them. Not requesting changes - use your judgment on any comments I made. Thanks!

File: secretmanager/snippets/snippets_test.py
Comment (glasnt) [reply secretmanager/snippets/snippets_test.py:69]: I originally tried `-> str`, but since this method doesn't return anything, that doesn't make sense. 

mypy says `The return type of a generator function should be "Generator" or one of its supertypes`

Adding just `Generator[str]` causes `"Generator" expects 3 type arguments, but 1 given`. Typing documentation for [Generator](https://docs.python.org/3/library/typing.html#typing.Generator) says you can use a triple for `[yield type, send type, return type]`, which is what I used. But that feels overly verbose

Reading further, I could just use `Iterator[str]` which looks cleaner, and mypy is happy.

File: secretmanager/snippets/snippets_test.py
Comment (glasnt) [reply secretmanager/snippets/snippets_test.py:87]: Same as above

File: secretmanager/snippets/consume_event_notification.py
Comment (glasnt) [reply secretmanager/snippets/consume_event_notification.py:23]: Yes, Any would be better here. I used `None` as that's what was passed by the test.

Comment (glasnt) [reply secretmanager/snippets/consume_event_notification.py]: Ah, it wouldn't be better. From Kokoro:

```
Dynamically typed expressions (typing.Any) are disallowed
```

**Comment by glasnt** [comment]
@engelke comments agreed, tests green. Not sure why I have merge permissions on this repo, but I'll let !me merge

**Comment by engelke** [comment]
Happy you have merge permission, but I'll go ahead now and merge it.

**Pull request merged**
